ext/pcre/tests/preg_match_non_capture.phpt

```diff
@@ -0,0 +1,25 @@
+--TEST--
+testing /n modifier in preg_match()
+--FILE--
+<?php
+
+preg_match('/.(.)./n', 'abc', $m);
+var_dump($m);
+
+preg_match('/.(?P<test>.)./n', 'abc', $m);
+var_dump($m);
+
+?>
+--EXPECT--
+array(1) {
+  [0]=>
+  string(3) "abc"
+}
+array(3) {
+  [0]=>
+  string(3) "abc"
+  ["test"]=>
+  string(1) "b"
+  [1]=>
```

Comment:
How is producing a numbered capture "passing" when that's what the `n` modifier is supposed to suppress? From the second call, I'd expect this output:
```
array(3) {
  [0]=>
  string(3) "abc"
  ["test"]=>
  string(1) "b"
}
```

Reply:
Fair enough, but that's a pity since Perl provides a separate hash for named groups, but PHP gets the numbered and named groups lumped in the same array.

Reply:
> Fair enough, but that's a pity since Perl provides a separate hash for named groups, but PHP gets the numbered and named groups lumped in the same array.

Not a problem there. PCRE behaves in this way, PHP is not creating this numbered entry. See PCRE2' documentation:

```
  PCRE2_NO_AUTO_CAPTURE

       If this option is set, it disables the use of numbered capturing paren-
       theses  in the pattern. Any opening parenthesis that is not followed by
       ? behaves as if it were followed by ?: but named parentheses can  still
       be used for capturing (and they acquire numbers in the usual way). This
       is the same as Perl's /n option.  Note that, when this option  is  set,
       references  to  capture  groups (backreferences or recursion/subroutine
       calls) may only refer to named groups, though the reference can  be  by
       name or by number.
```

Reply:
Yes, I understood that PCRE returns matches with both numbered and named references, but surely it has always been PHP's choice to put both of those reference sets into the one array and not provide a way to conditionally omit/separate the numbered set, has it not? This is unlike Perl which puts the named set into its own hash:
```sh
echo abc | perl -ne '/(.)(?<x>.)(.)/n; $s = keys %+; print "$s\n"'
```